client/landing/stepper/declarative-flow/internals/hooks/use-step-navigation-with-tracking/index.ts

```diff
@@ -92,6 +99,22 @@ export const useStepNavigationWithTracking = ( {
 					stepNavigation.exitFlow?.( to );
 				},
 			} ),
+			/**
+			 * If the `previousStep` is defined in the store, it's a solid proxy to guess that we navigated at least once via Stepper's React Router.
+			 * If the flow doesn't define a `goBack` handler, and `previousStep` is defined, we can just go history.back() and we'll remain in the flow.
+			 * But if `previousStep` is not defined, and the flow doesn't define a `goBack` handler, we should return undefined so the StepContainer doesn't render a back button.
```

Comment:
Is it possible now for a flow to not define a `goBack` handler, since it will always inherit one?

Reply:
Is it worth offering a hybrid option so flows can only override the behaviour in specific situations? That would make it easier for flows to fall through to the default in most cases, and exceptions would not require the flow to implement handling for all possible cases.

Reply:
We can easily check the return value of the `goBack` from the flow and use that to decide `flow.goBack?.() ?? defaultGoBack()` but at this point, especially after [your comment](https://github.com/Automattic/wp-calypso/pull/101550#issuecomment-2736042430), I'm convinced we should remove going back from the flow level. It should always be aligned with the browser's back button.

Reply:
Actually, can this be a side-effect or secondary helper? In my mind, the only thing the flow should be doing is interacting with the flow state, largely in cases where the user might end up leaving the flow.

Reply:
I don't disagree, but I'm increasingly dispassionate about increasing the API surface area. Can we wait until we need that?

Reply:
I can think of an edge case that we don't have a good solution for: what happens when you go back to the `processing` step?

At present, users often get stuck because the original step that triggered an async action is the step before the processing step, and we haven't (re)triggered that original action.

cc @gabrielcaires as he has been thinking about this as a general Stepper issue/gap.

Reply:
Processing and site creation steps should replace history. Like [this](https://github.com/Automattic/wp-calypso/blob/trunk/client/landing/stepper/declarative-flow/flows/onboarding/onboarding.ts#L381-L384).